### PR TITLE
CLI Cluster Storage Group Add via RPC

### PR DESCRIPTION
**Pull request by dvaldivia**

### CODE REVIEW

File: portal/api-private-cluster.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"context"
+	"fmt"
+
+	"github.com/minio/m3/cluster"
+	pb "github.com/minio/m3/portal/stubs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// AddTenant rpc to add a new tenant and it's first user
```
Comment (cesnietor) [inline portal/api-private-cluster.go]: doc update?

File: portal/api-private-cluster.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+func (ps *privateServer) ClusterScSgAdd(ctx context.Context, in *pb.StorageGroupAddRequest) (*pb.StorageGroupAddResponse, error) {
+	appCtx, err := cluster.NewEmptyContext()
+	if err != nil {
+		return nil, status.New(codes.Internal, "An internal error happened").Err()
+	}
+
+	var name *string
+	if in.Name != "" {
+		name = &in.Name
+	}
```
Comment (cesnietor) [inline portal/api-private-cluster.go:39]: `name := in.GetName()` it does pretty much what you are doing.

**Review by cesnietor — CHANGES_REQUESTED** [review]

File: portal/api-private-cluster.go (hunk 2 above)
Comment (dvaldivia) [reply portal/api-private-cluster.go:39]: I need an empty reference if the name was not passed `in.Name` and `in.GetName()` will return "" if the string was not passed

File: portal/api-private-cluster.go (hunk 1 above)
Comment (dvaldivia) [reply portal/api-private-cluster.go]: done

**Review by cesnietor — APPROVED** [review]

**Review by cesnietor — APPROVED** [review]

**Pull request merged**